cli/src/commands/branch.rs

```diff
@@ -759,24 +759,32 @@ fn cmd_branch_list(
             if local_target.is_present() && !synced {
                 let remote_added_ids = remote_ref.target.added_ids().cloned().collect_vec();
                 let local_added_ids = local_target.added_ids().cloned().collect_vec();
-                let remote_ahead_count =
+                let (remote_ahead_lower, remote_ahead_upper) =
                     revset::walk_revs(repo.as_ref(), &remote_added_ids, &local_added_ids)?
-                        .iter()
-                        .count();
-                let local_ahead_count =
+                        .count_estimate();
+                let (local_ahead_lower, local_ahead_upper) =
                     revset::walk_revs(repo.as_ref(), &local_added_ids, &remote_added_ids)?
-                        .iter()
-                        .count();
-                let remote_ahead_message = if remote_ahead_count != 0 {
-                    Some(format!("ahead by {remote_ahead_count} commits"))
-                } else {
-                    None
-                };
-                let local_ahead_message = if local_ahead_count != 0 {
-                    Some(format!("behind by {local_ahead_count} commits"))
-                } else {
-                    None
-                };
+                        .count_estimate();
+                let remote_ahead_message =
+                    if remote_ahead_upper.is_some_and(|x| x == remote_ahead_lower) {
+                        if remote_ahead_lower == 0 {
+                            None
+                        } else {
+                            Some(format!("ahead by {remote_ahead_lower} commits"))
+                        }
+                    } else {
+                        Some(format!("ahead by at least {remote_ahead_lower} commits"))
+                    };
```

Comment:
I think you can simplify this with some pattern matching:

```rust
let remote_ahead_message =
  if remote_ahead_upper == Some(0) {
    None
  } else if remote_ahead_upper == Some(remote_ahead_lower) {
    Some(format!("ahead by {remote_ahead_lower} commits"))
  } else {
    Some(format!("ahead by at least {remote_ahead_lower} commits"))
  }
}
```

Reply:
You can also write it like this:

```rust
let remote_ahead_message = match remote_ahead_upper {
    (Some(0)) => None,
    (Some(upper)) if upper == remote_ahead_lower => { ... }
    _ => { ... }
}
```

Up to you if that's better than the `if`/`else`s, though.

Reply:
Thanks. I used arxanas's suggestion.